test/uk/gov/hmrc/transitmovements/controllers/DeparturesControllerSpec.scala

```diff
@@ -104,15 +145,24 @@ class DeparturesControllerSpec
         val elementNotFoundXml: NodeSeq =
           <CC015C></CC015C>
 
+        when(mockXmlParsingService.extractDeclarationData(any[Source[ByteString, _]]))
+          .thenReturn(EitherT.leftT(ParseError.NoElementFound("messageSender")))
+
+        when(mockTemporaryFileCreator.create()).thenReturn(SingletonTemporaryFileCreator.create())
+
         val request = fakeRequestDepartures(POST, elementNotFoundXml)
 
-        val result = app.injector.instanceOf[DeparturesController].post()(request)
+        val result = route(app, request).value
 
         status(result) mustBe BAD_REQUEST
         contentAsJson(result) mustBe Json.obj(
           "code"    -> "BAD_REQUEST",
           "message" -> "Element messageSender not found"
         )
+
+        reset(mockTemporaryFileCreator)
+        reset(mockXmlParsingService)
+        reset(mockDeparturesService)
```

Comment:
There is a beforeAndAfter testing class you can inherit from that will let you override a reset method that you can put these resets in rather than resetting them in every test